Sources/X509/CertificatePrivateKey.swift

```diff
@@ -202,3 +202,69 @@ extension Certificate.PrivateKey {
         }
     }
 }
+
+@available(macOS 11.0, iOS 14, tvOS 14, watchOS 10, *)
+extension Certificate.PrivateKey {
+    @usableFromInline
+    static let pemDiscriminatorForRSAPrivateKey = "RSA PRIVATE KEY"
+    
+    @usableFromInline
+    static let pemDiscriminatorForSEC1PrivateKey = "EC PRIVATE KEY"
+    
+    @usableFromInline
+    static let pemDiscriminatorForPKCS8PrivateKey = "PRIVATE KEY"
+    
+    @inlinable
+    public init(pemEncoded: String) throws {
+        try self.init(pemDocument: PEMDocument(pemString: pemEncoded))
+    }
+    
+    @inlinable
+    public init(pemDocument: PEMDocument) throws {
+        switch pemDocument.discriminator {
+        case Self.pemDiscriminatorForRSAPrivateKey:
+            self = try .init(_CryptoExtras._RSA.Signing.PrivateKey.init(derRepresentation: pemDocument.derBytes))
+            
+        case Self.pemDiscriminatorForSEC1PrivateKey:
+            let parsed = try SEC1PrivateKey(derEncoded: pemDocument.derBytes)
+            self = try .init(algorithm: parsed.algorithm, rawEncodedPrivateKey: parsed.privateKey.bytes)
+            
+        case Self.pemDiscriminatorForPKCS8PrivateKey:
+            let parsed = try PKCS8PrivateKey(derEncoded: pemDocument.derBytes)
+            self = try .init(algorithm: parsed.algorithm, rawEncodedPrivateKey: parsed.privateKey.privateKey.bytes)
+            
+        default:
+            throw ASN1Error.invalidPEMDocument(
+                reason: "PEMDocument has incorrect discriminator \(pemDocument.discriminator). Expected \(Self.pemDiscriminatorForPKCS8PrivateKey), \(Self.pemDiscriminatorForSEC1PrivateKey) or \(Self.pemDiscriminatorForRSAPrivateKey) instead"
+            )
+        }
+    }
+    
+    @inlinable
+    init(algorithm: AlgorithmIdentifier?, rawEncodedPrivateKey: ArraySlice<UInt8>) throws {
+        switch algorithm {
+        case .some(.ecdsaP256):
+            self = try .init(P256.Signing.PrivateKey(rawRepresentation: rawEncodedPrivateKey))
+        case .some(.ecdsaP384):
+            self = try .init(P384.Signing.PrivateKey(rawRepresentation: rawEncodedPrivateKey))
+        case .some(.ecdsaP521):
+            self = try .init(P521.Signing.PrivateKey(rawRepresentation: rawEncodedPrivateKey))
+        default:
```

Comment:
I think we are missing support for RSA keys here but I couldn't find the algorithm identifier from the RFCs or CryptoKit code.
cc @Lukasa

Reply:
The algorithm identifier should be `.rsaEncryption`, I believe.